src/VscPreviewExtension/src/extension.ts

```diff
@@ -0,0 +1,270 @@
+        }
+    });
+
+    workspace.onDidChangeTextDocument(event => {
+        if (isMarkdownFile(event.document)) {
+            documentUri = getMarkdownUri(event.document.uri);
+            dfmProcess.callDfm();
+        }
+    });
+
+    workspace.onDidChangeConfiguration(() => {
+        workspace.textDocuments.forEach(document => {
+            if (document.uri.scheme === 'markdown') {
+                documentUri = documentUri;
+                dfmProcess.callDfm();
+            }
+        });
+    });
+}
+
+// Check the file type
+function isMarkdownFile(document: TextDocument) {
+    // Prevent processing of own documents
+    return document.languageId === 'markdown' && document.uri.scheme !== 'markdown';
+}
+
+function getMarkdownUri(uri: Uri) {
+    return uri.with({ scheme: 'markdown', path: uri.path + '.rendered', query: uri.toString() });
+}
+
+function showPreview(dfmPreview: PreviewCore, uri?: Uri, sideBySide: boolean = false) {
+    if (window.activeTextEditor.document.languageId !== 'markdown') {
+        window.showErrorMessage("This is not a markdown file!");
+        return;
+    }
+    dfmPreview._isFirstTime = true;
+    let resource = uri;
+    if (!(resource instanceof Uri)) {
+        if (window.activeTextEditor) {
+            resource = window.activeTextEditor.document.uri;
+        }
+    }
+
+    if (!(resource instanceof Uri)) {
+        if (!window.activeTextEditor) {
+            // This is most likely toggling the preview
+            return commands.executeCommand('markdown.showSource');
+        }
+        // Nothing found that could be shown or toggled
+        return;
+    }
+
+    let thenable = commands.executeCommand('vscode.previewHtml',
+        getMarkdownUri(resource),
+        getViewColumn(sideBySide),
+        `Preview '${path.basename(resource.fsPath)}'`);
+
+    documentUri = getMarkdownUri(resource);
+    dfmPreview.callDfm();
+    return thenable;
+}
+
+function getViewColumn(sideBySide): ViewColumn {
+    const active = window.activeTextEditor;
+    if (!active) {
+        return ViewColumn.One;
+    }
+
+    if (!sideBySide) {
+        return active.viewColumn;
+    }
+
+    switch (active.viewColumn) {
+        case ViewColumn.One:
+            return ViewColumn.Two;
+        case ViewColumn.Two:
+            return ViewColumn.Three;
+    }
+
+    return active.viewColumn;
+}
+
+function showSource(mdUri: Uri) {
+    if (!mdUri) {
+        return commands.executeCommand('workbench.action.navigateBack');
+    }
+
+    const docUri = Uri.parse(mdUri.query);
+
+    for (let editor of window.visibleTextEditors) {
+        if (editor.document.uri.toString() === docUri.toString()) {
+            return window.showTextDocument(editor.document, editor.viewColumn);
+        }
+    }
+
+    return workspace.openTextDocument(docUri).then(doc => {
+        return window.showTextDocument(doc);
+    });
+}
+
+// Create a child process(DfmRender) by '_spawn' to render a html
+class PreviewCore {
```

Comment:
the global varilable access make it a little complex

---
In reply to: [75244578](https://github.com/dotnet/docfx/pull/616#discussion_r75244578) [](ancestors = 75244578)